test/test_dists.jl

```diff
@@ -504,13 +504,17 @@ function test_pairwise(dist, x, y, T)
         for j = 1:nx, i = 1:nx
             rxx[i, j] = dist(x[:, i], x[:, j])
         end
+        vecx = collect(x[:, i] for i in 1:nx)
+        vecy = collect(y[:, i] for i in 1:ny)
         # As earlier, we have small rounding errors in accumulations
         @test pairwise(dist, x, y, dims=2) ≈ rxy
         @test pairwise(dist, x, dims=2) ≈ rxx
         @test pairwise(dist, x, y, dims=2) ≈ rxy
         @test pairwise(dist, x, dims=2) ≈ rxx
         @test pairwise(dist, permutedims(x), permutedims(y), dims=1) ≈ rxy
         @test pairwise(dist, permutedims(x), dims=1) ≈ rxx
+        @test pairwise(dist, vecx, vecy) ≈ rxy
+        @test pairwise(dist, vecx) ≈ rxx
```

Comment:
This would also allow computing the pairwise distance matrix for number vectors, by computing the pairwise scalar distances. This currently errors:
```julia
julia> using Distances

julia> a = rand(5);

julia> pairwise(Euclidean(), a, a)
ERROR: MethodError: no method matching pairwise(::Euclidean, ::Array{Float64,1}, ::Array{Float64,1})
```
Allowing that is only consistent, because we do have, say, `Euclidean()` for both numbers and arrays of numbers. But then we should test it, maybe something like:
```suggestion
        @test pairwise(dist, vecx) ≈ rxx
        vec1 = x[:, 1]; vec2 = y[:, 1]
        @test pairwise(dist, vec1, vec2) ≈ dist.(vec1, permutedims(vec2))
```

Reply:
Aha, I see, this was indeed proposed in #165, as you mentioned. 👍